packages/rest/src/session/AbstractSession.ts

```diff
@@ -217,10 +217,17 @@ export abstract class AbstractSession {
             const authArr = auth.split(";");
             // see each field in the cookie, e/g. Path=/; Secure; HttpOnly; LtpaToken2=...
             authArr.forEach((element: string) => {
-                // if we match requested token type, save it off for its length
-                if (element.indexOf(this.mISession.tokenType) === 0) {
-                    // parse off token value, minus LtpaToken2= (as an example)
-                    this.ISession.tokenValue = element.substr(this.ISession.tokenType.length + 1, element.length);
+                // if element begins with tokenType, extract full tokenType and tokenValue.
+                if (element.indexOf(this.ISession.tokenType) === 0) {
+                    // parse off token value, splitting element at first "=".
+                    const split = element.indexOf("=");
+                    if (split >= 0) {
+                        this.ISession.tokenType  = element.substring(0, split);
+                        this.ISession.tokenValue = element.substring(split + 1);
+                    }
+                    else {
+                        this.ISession.tokenValue = "";
```

Comment:
This may introduce a breaking change if someone has a condition checking whether `tokenValue` is null or undefined. Could the `else` clause here be removed to leave `tokenValue` undefined if not present?

Reply:
Certainly. I'll include that change along with the unit test you suggested.

Reply:
Requested changes have been implemented.

Reply:
Thanks for adding the test and addressing the comments! 👍 

To really illustrate the problem, could you possibly use a random tokenType and non-realistic tokenValue (something that would result in `tokenType = "cool" ; tokenValue = "beans"`) and isntead of using snapshots, explicitly expect the token value to be what you set it to (`"beans"` for example).

The reason behind the request to avoid snapshots for tests catching breaking changes is that someone can easily overlook a snapshot value with the `--updateSnapshot` flag for `jest`. 
For example, I copied the same exact test (with `tokenType="LtpaToken"` instead of `LtpaToken2`) and the only thing that changed was the initial equal sign (`=7KM...` ). However it reported that the tests passed since there was no snapshot already for it in master.

Reply:
Unit test has been updated to reflect zFernand0's suggestion.

Reply:
Thanks a lot @Tulsagrammer.
Were you able to delete the snapshot? I believe I can still see 4 files changed (and one of them being the snapshot addition)

Reply:
@zFernand0 just deleted the snapshot the earlier unit test created. Now showing three files.